data/awsrds/v0/main.go

```diff
@@ -0,0 +1,448 @@
+		client, err = newClientIAM(e)
+	} else {
+		return nil, fmt.Errorf("some fields are missing, please fill in the required fields")
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("error creating SQL client: %v", err)
+	}
+	defer client.Close()
+
+	for _, input := range inputs {
+		var output *structpb.Struct
+
+		switch e.Task {
+		case taskUpsert:
+			tableName := input.GetFields()["table-name"].GetStringValue()
+
+			var err error
+			if ListValue, ok := input.GetFields()["data"].GetKind().(*structpb.Value_ListValue); ok {
+				for _, data := range ListValue.ListValue.Values {
+					err = UpsertUser(client, &tableName, data.GetStructValue())
+					if err != nil {
+						return nil, err
+					}
+				}
+			} else {
+				// Unmarshal the JSON data from the protobuf into structpb.Struct
+				data := input.GetFields()["data"].GetStructValue()
+
+				err = UpsertUser(client, &tableName, data)
+			}
+
+			if err != nil {
+				return nil, err
+			}
+
+			output = &structpb.Struct{Fields: map[string]*structpb.Value{
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
+		case taskSelect:
+			tableName := input.GetFields()["table-name"].GetStringValue()
+
+			// Unmarshal the JSON data from the protobuf into structpb.Struct
+			criteria := input.GetFields()["criteria"].GetStructValue()
+			from := input.GetFields()["from"].GetNumberValue()
+			to := input.GetFields()["to"].GetNumberValue()
+
+			outputList, err := SelectUser(client, &tableName, criteria, int(from), int(to))
+			if err != nil {
+				return nil, err
+			}
+			output = &structpb.Struct{Fields: map[string]*structpb.Value{
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}},
+				"rows":   {Kind: &structpb.Value_ListValue{ListValue: outputList}},
+			}}
+
+		case taskDelete:
+			tableName := input.GetFields()["table-name"].GetStringValue()
+
+			// Unmarshal the JSON data from the protobuf into structpb.Struct
+			criteria := input.GetFields()["criteria"].GetStructValue()
+
+			err := DeleteUser(client, &tableName, criteria)
+			if err != nil {
+				return nil, err
+			}
+
+			output = &structpb.Struct{Fields: map[string]*structpb.Value{
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
+		case taskQuery:
+			query := input.GetFields()["query"].GetStringValue()
+
+			err := QueryUser(client, query)
+			if err != nil {
+				return nil, err
+			}
+
+			output = &structpb.Struct{Fields: map[string]*structpb.Value{
+				"status": {Kind: &structpb.Value_StringValue{StringValue: "Successfully executed SQL statement"}}}}
+		default:
+			return nil, fmt.Errorf("task %s not supported", e.Task)
+		}
+		outputs = append(outputs, output)
+	}
+	return outputs, nil
+}
+
+func UpsertUser(client *sqlx.DB, tableName *string, data *structpb.Struct) error {
+	// Begin constructing SQL statement
+	sqlStatement := "INSERT INTO " + *tableName + " ("
```

Comment:
Question
I happened to realise that the different database could need different command.
Could you help us check if the commands of the db you support here, postgres, oracle, ..., are same?

Reply:
@chuang8511 

Since all of the mentioned databases use SQL, they should have the same, if not very similar, commands. The differences lie in how they manage the data within the databases themselves. Although there are some differences in specific commands for each engine, basic and simple tasks like SELECT, DELETE, and UPSERT are mostly the same.

[https://aws.amazon.com/compare/the-difference-between-mysql-vs-postgresql/#:~:text=In%20a%20MySQL%20database%2C%20you,complex%20SQL%20statements%20using%20functions.](url)